### DISCUSSION TITLE
[Dialog] and "portalled" content

### DISCUSSION THREAD

**Discussion by TrevorRice** [radix-ui/primitives#1012] [Help]
Hi all!

Was curious to get your thoughts on components passed to `Dialog` that leverage portals.

[Here's an example](https://codesandbox.io/s/optimistic-morse-koqn7?file=/App.js) (albeit a bit contrived) that places Reach UI's [`Listbox`](https://reach.tech/listbox) inside of `Dialog`. While very specific, I would imagine the below could be seen in other similar portal usage.

I'm noticing two things:

1. The listbox is closed immediately after opening. I'm guessing this may have to do with the `Dialog`'s trapping logic?
2. Assuming the above could be resolved, the content placed in the portal (in this case the listbox options) would be hidden from assistive technologies since the portal container has `aria-hidden="true"` applied to it. Is there a way to exclude elements outside of the `Dialog`'s DOM tree from the "hidden" logic? FWIW, Reach's own `Dialog` has the [same issue](https://github.com/reach/reach-ui/issues/870).
![Screen Shot 2021-12-02 at 9 43 42 AM](https://user-images.githubusercontent.com/3644075/144454588-1e07b5ee-6c47-4337-b160-4c49e0860a6b.png)

Wasn't sure if this is an expected/supported use-case, but I thought I'd bring it up to get your thoughts.

Amazing work on Radix by the way!

**Comment by jjenzz** [comment]
Hi @TrevorRice 👋 

We have a semi [related issue](https://github.com/radix-ui/primitives/issues/922) on this. We've discussed [updating our focus scope logic to consider the React tree](https://github.com/radix-ui/primitives/issues/922#issuecomment-947775438) instead and I'm keen to give it a go, it's just finding the time at the moment. 

In the meantime, you can try any of these workarounds https://github.com/radix-ui/primitives/issues/922#issuecomment-947735302

> Amazing work on Radix by the way!

Thank you 💛

**Reply by TrevorRice** [reply]
Appreciate the response @jjenzz!

> We've discussed updating our focus scope logic to consider the React tree instead

That's awesome! Would this account for the `aria-hidden` logic as well? [From what I've been able to gather](https://github.com/radix-ui/primitives/blob/main/packages/react/dialog/src/Dialog.tsx#L237-L241), this lives outside of and is decoupled from the focus scope.

> ... I'm keen to give it a go, it's just finding the time at the moment.

I hear that 🙂

> In the meantime, you can try any of these workarounds #922 (comment)

Thank you!

**Reply by jjenzz** [reply]
> That's awesome! Would this account for the `aria-hidden` logic as well?

Ah yes, any solution would need to solve that too 👍
